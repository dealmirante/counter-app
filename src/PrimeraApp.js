import React from 'react'; //imr tab
import Proptypes from 'prop-types';
// <> es la forma abreviada de usar fragment sin tener que importarlo

const PrimeraApp = ({saludo, subtitulo})=> {

    return (
        <>             
            <h1>{saludo}</h1>
            <h2>Esto es un parrafo</h2>
            <p>{subtitulo}</p>
        </>
    ) 
}

PrimeraApp.propTypes = {
    saludo: Proptypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: "Soy un subtitulo por defecto (defaultProps)"
}

export default PrimeraApp;