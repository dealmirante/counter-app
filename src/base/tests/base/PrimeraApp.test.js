import React from 'react';
//import { render } from '@testing-library/react';
import PrimeraApp from '../../../PrimeraApp';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';


describe('pruebas en el componente PrimeraApp', () => {
    // ctrl + C
    // test('Debe mostrar el mensaje "Hola, soy Gabriel" ', () => {
    //     const saludo = "Hola, soy Gabriel";
    //     const {getByText} = render(<PrimeraApp saludo = {saludo}/>);
    //     expect( getByText(saludo) ).toBeInTheDocument();        
    // });
    test('Debe de mostrar el componente <PrimeraApp/> correctamente', () => {
        // shallow funciona como render
        const saludo = "Hola, soy Gabriel";
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);
        // al ejecutar este expect me genera una carpeta _snapshots_
        // npm install --save-dev enzyme-to-json
        expect(wrapper).toMatchSnapshot();
        
        
    })
    test('Debe mostrarse el subtitulo enviado por props ', () => {
        const saludo = "Hola, soy Gabriel";
        const subTitulo = "Soy un subtitulo por defecto (defaultProps)";        

        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo} 
                subtitulo={subTitulo}
            />
        );

        // el find actua como el querySelector
        const textoParrafo = wrapper.find('p').text();

        expect(subTitulo).toBe(textoParrafo);


        
    })
    
    
    
});
