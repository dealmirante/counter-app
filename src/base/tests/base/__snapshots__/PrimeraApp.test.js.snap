// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente PrimeraApp Debe de mostrar el componente <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Gabriel
  </h1>
  <h2>
    Esto es un parrafo
  </h2>
  <p>
    Soy un subtitulo por defecto (defaultProps)
  </p>
</Fragment>
`;
